### PR TITLE
Fix `pytest.mark.usefixtures()` without arguments silently erroring

### LINKED ISSUE #12426: `pytest.mark.usefixtures()` without argument gets silently ignored
Someone in my pytest training accidentally did:

```python
import pytest

@pytest.mark.usefixtures()
def test_fixt():
    pass
```

but forgot to pass a fixture name. Because we do:

https://github.com/pytest-dev/pytest/blob/0070151c630b30e38381f1a3a00e263de4665e26/src/_pytest/fixtures.py#L1536-L1539

That results in the mark being silently ignored. IMHO, this should result in a warning or even error instead.

I think this would be a great beginner issue for someone that never contributed to pytest before. Maybe someone at the upcoming sprint wants to take a look?

### PR DESCRIPTION
**Pull request by daara-s**
Follow up to #12439.

Closes #12426
Closes #12439

### CODE REVIEW

File: testing/python/fixtures.py [hunk 1]
```diff
@@ -1435,6 +1435,19 @@ def test_two(self):
...
+            """
+            import pytest
+
+            @pytest.mark.usefixtures()
+            def test_one():
+                assert 1 == 1
+        """
+        )
+        reprec = pytester.inline_run()
+        reprec.assertoutcome(failed=1)
```
Comment (nicoddemus) [inline testing/python/fixtures.py:1449]: This is good, but I feel we should test the actual message, using `pytest.runpytest()` and `fnmatch_lines()`.

**Review by nicoddemus — CHANGES_REQUESTED** [review]

**Review by nicoddemus — APPROVED** [review]
Thanks @daara-s for picking this up, and also thanks to @JiajunXu2 for the initial PR. 👍

**Comment by The-Compiler** [comment]
Oh, that's unfortunate... Like you say I suppose pytest-bdd could do the decorating manually and conditionally, in the same way as just [below the linked code snippet](https://github.com/pytest-dev/pytest-bdd/blob/93e446eff667b9f4f450ea239d19a19e335590c7/src/pytest_bdd/scenario.py#L284-L290):

```python
        example_parametrizations = collect_example_parametrizations(templated_scenario)
        if example_parametrizations is not None:
            # Parametrize the scenario outlines
            scenario_wrapper = pytest.mark.parametrize(
                "_pytest_bdd_example",
                example_parametrizations,
            )(scenario_wrapper)
```

cc @youtux @jsa34 as a heads-up.

**Comment by daara-s** [comment]
@The-Compiler [this what you mean](https://github.com/pytest-dev/pytest-bdd/pull/746)?

**Comment by daara-s** [comment]
Got ahead of myself with the revert. Still need to bump the plugin pin.

@youtux what's the release cadence for pytest-bdd?

**Comment by youtux** [comment]
We don't really have a schedule, just whenever we think it's needed.

**Comment by nicoddemus** [comment]
Thanks @daara-s and @JiajunXu2 for the work here! 👍

**Comment by JiajunXu2** [comment]
Sorry for stepping away from the pr, I've been a bit busy lately. Thank you @daara-s for finishing it, and everyone for the support!

**Pull request merged**